server/src/main/java/org/opensearch/index/store/remote/filecache/FullFileCachedIndexInput.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.store.remote.filecache;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.lucene.store.AlreadyClosedException;
+import org.apache.lucene.store.IndexInput;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.HashSet;
+import java.util.Set;
+
+public class FullFileCachedIndexInput extends FileCachedIndexInput {
+    private static final Logger logger = LogManager.getLogger(FullFileCachedIndexInput.class);
+    private final Set<FullFileCachedIndexInput> clones;
+
+    public FullFileCachedIndexInput(FileCache cache, Path filePath, IndexInput underlyingIndexInput) {
+        this(cache, filePath, underlyingIndexInput, false);
+    }
+
+    public FullFileCachedIndexInput(FileCache cache, Path filePath, IndexInput underlyingIndexInput, boolean isClone) {
+        super(cache, filePath, underlyingIndexInput, isClone);
+        clones = new HashSet<>();
+    }
+
+    @Override
+    public FullFileCachedIndexInput clone() {
+        FullFileCachedIndexInput clonedIndexInput = new FullFileCachedIndexInput(cache, filePath, luceneIndexInput.clone(), true);
+        cache.incRef(filePath);
+        clones.add(clonedIndexInput);
+        return clonedIndexInput;
+    }
+
+    @Override
+    public IndexInput slice(String sliceDescription, long offset, long length) throws IOException {
+        if (offset < 0 || length < 0 || offset + length > this.length()) {
+            throw new IllegalArgumentException(
+                "slice() "
+                    + sliceDescription
+                    + " out of bounds: offset="
+                    + offset
+                    + ",length="
+                    + length
+                    + ",fileLength="
+                    + this.length()
+                    + ": "
+                    + this
+            );
+        }
+        IndexInput slicedLuceneIndexInput = luceneIndexInput.slice(sliceDescription, offset, length);
+        FullFileCachedIndexInput slicedIndexInput = new FullFileCachedIndexInput(cache, filePath, slicedLuceneIndexInput, true);
+        clones.add(slicedIndexInput);
+        cache.incRef(filePath);
+        return slicedIndexInput;
+    }
+
+    @Override
+    public void close() throws IOException {
+        if (!closed) {
+            if (isClone) {
+                cache.decRef(filePath);
+            }
+            clones.forEach(indexInput -> {
+                try {
+                    indexInput.close();
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
```

Comment:
Instead of using `IOUtils.closeWhileHandlingException(indexInput)`, quietly consumed the exception and added  trace log in the catch block so that it can be used while debugging